src/output-json-ssh.c

```diff
@@ -75,25 +75,25 @@ static int JsonSshLogger(ThreadVars *tv, void *thread_data, const Packet *p,
         return 0;
     }
 
-    json_t *js = CreateJSONHeader(p, LOG_DIR_FLOW, "ssh", NULL);
+    JsonBuilder *js = CreateEveHeaderWithTxId(p, LOG_DIR_FLOW, "ssh", NULL, tx_id);
     if (unlikely(js == NULL))
         return 0;
 
-    JsonAddCommonOptions(&ssh_ctx->cfg, p, f, js);
+    EveAddCommonOptions(&ssh_ctx->cfg, p, f, js);
 
     /* reset */
     MemBufferReset(aft->buffer);
 
-    json_t *tjs = rs_ssh_log_json(txptr);
-    if (unlikely(tjs == NULL)) {
-        free(js);
+    jb_open_object(js, "ssh");
+    if (!rs_ssh_log_json(txptr, js)) {
+        jb_free(js);
         return 0;
     }
-    json_object_set_new(js, "ssh", tjs);
+    if (jb_close(js)) {
```

Comment:
When should we test the return value of `jb_close` ?
Sometimes, it is checked, and sometimes not...

Reply:
In Rust I'd say always, as its easy to do everywhere with '?'.  In C we're not checking it really.  Not for a good reason, but more for the reason that we didn't check libjansson responses either, all of which can fail in more or less the same place as a JsonBuilder.

The bigger questions I suppose is what to do on error? It will most likely be a memory allocation panic as we go out of our way to handle UTF8 encoding errors.  I suppose JsonError::Invalid state could happen as well, but thats likely a programming error (should add a debug validate on that - just inside jsonbuilder)

Reply:
Ok, so removing the check for now